csrc/ops/composite.cpp

```diff
@@ -389,38 +389,7 @@ TensorView* matmul(TensorView* tv_a, TensorView* tv_b) {
       " and ",
       tv_b->dtype());
 
-  // Check for K=1 i.e. reduction of broadcast. In these cases we don't need a
```

Comment:
So we are actually slower than aten for this case? I wonder what their kernel looks like and how we could improve the pointwise scheduler in these simple broadcast cases.

Reply:
No. 
The dot product is somewhat tricky, nvFuser is a little faster (around 2% for the test cases I tried) for very large K (order of 10^8) and slower for smaller K (order of 10^5). For smaller K, the kernel time is under 10us, and the difference between eager and nvFuser can be upto 25%.

In an earlier discussion, we decided to use `MatmulOp` ubiquitously, since these are corner cases and it would be easier to debug any performance regressions if we have use ATen for all cases.

We can revisit this, if you think we should handle this case separately.
CC: @kevinstephano

Reply:
Agreed. So far, the models are not using matmul for these cases afaik. If such a case appears, then we could potentially use `fuse_matmul` option here to enable translation.

Reply:
> If such a case appears, then we could potentially use `fuse_matmul` option here to enable translation.

That's not so straightforward though. We would be accepting the segment with the Matmul scheduler then, when we would need it to be accepted by something else like pointwise, transpose, reduction, etc. This is why it's better to translate as early as possible so that we can properly segment. However, as you say, until we find this use case in the wild we should probably not spend too much energy optimizing it. It is, after all, a translation that can also happen further up the stack like in thunder or could be done by the user.

Reply:
Yes, we cannot do it after segmentation. I was proposing doing that right here, when we are creating a `MatmulOp`. So same as the current special cases guarded by this enable option.

However, for now, let us move with removing these cases and reevaluate when a relevant fusion comes up.

Reply:
> I was proposing doing that right here, when we are creating a `MatmulOp`

I am not sure I understand how that'd be different from the current behavior. The `fuse_matmul` option shouldn't affect whether we do that translation since there would no longer be a `MatmulPattern` in the fusion if you skip the `MatmulOp` for this case. Anyway, as you say let's proceed with removing it :grin: .